File: app/src/main/java/com/example/flightstats/data/Flight.java
package com.example.flightstats.data;

import androidx.annotation.NonNull;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "flights")
public class Flight {
    @PrimaryKey(autoGenerate = true)
    public int id;

    public String date;        // YYYY-MM-DD
    public String origin;      // IATA code e.g. "AMS"
    public String destination; // IATA code e.g. "JFK"
    public double distance;    // km

    // Optional user-entered details
    public String flightNumber; // e.g. "KL871"
    public String airline;      // e.g. "KLM"
    public String seat;         // e.g. "23A"
    public String seatClass;    // "Economy", "Business", "First", "Premium Economy"
    public String notes;        // free-form notes
}

File: app/src/main/java/com/example/flightstats/data/Flight.kt
package com.example.flightstats.data

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "flights")
data class Flight(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val date: String?,         // YYYY-MM-DD
    val origin: String?,        // IATA code e.g. "AMS"
    val destination: String?,   // IATA code e.g. "JFK"
    val distance: Double,      // km
    val flightNumber: String?, // e.g. "KL871"
    val airline: String?,      // e.g. "KLM"
    val seat: String?,         // e.g. "23A"
    val seatClass: String?,    // "Economy", "Business", "First", "Premium Economy"
    val notes: String?,        // free-form notes
    val departureTime: String?, // e.g. "14:30"
    val arrivalTime: String?   // e.g. "21:45"
)
